src/notebook_utils/notebook_utils.py

```diff
@@ -0,0 +1,41 @@
+import itables.options as opt
+from itables import init_notebook_mode, show
+from pandas import DataFrame
+
+
+def display_df(
+        df: DataFrame,
+        layout: dict = None,
+        buttons: list = None,
+        length_menu: list = None
+) -> None:
+    """
+    Display a pandas DataFrame using itables.
+    iTables project page: https://github.com/mwouts/itables
+
+    :param df: a pandas DataFrame
+    :param layout: layout options, refer to https://datatables.net/reference/option/layout
+    :param buttons: buttons options, options refer to https://datatables.net/reference/button/
+    :param length_menu: length menu options
+    :return:
+    """
+    # initialize itables for the notebook
+    init_notebook_mode(all_interactive=False)
+
+    # set default values if options are not provided
+    default_layout = {
+        "topStart": "search",
+        "topEnd": "buttons",
+        "bottomStart": "pageLength",
+        "bottomEnd": "paging",
+        "bottom2Start": "info"
+    }
+    default_buttons = ["csvHtml5", "excelHtml5", "print"]
+    default_length_menu = [5, 10, 20]
+
+    layout = layout or default_layout
+    buttons = buttons or default_buttons
+    length_menu = length_menu or default_length_menu
+
+    opt.layout = layout
```

Comment:
This looks like a race condition theoretically

Reply:
👍 added a lock to prevent race condition.

Reply:
a `lock = RLock()` outside of `display_df`?

Reply:
yeah, just at the package level should work

Reply:
👍

Reply:
:+1: